src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -621,8 +622,51 @@ class generates the expected code. Use the gocean API.
     clauses = ""
     if grainsize is not None:
         clauses = " grainsize({0})".format(grainsize)
+        if nogroup:
+            clauses = clauses + ", nogroup"
     if num_tasks is not None:
         clauses = " num_tasks({0})".format(num_tasks)
+        if nogroup:
+            clauses = clauses + ", nogroup"
+
+    assert (
+        "    !$omp parallel default(shared), private(i,j)\n" +
+        "      !$omp master\n" +
+        "      !$omp taskloop{0}\n".format(clauses) +
+        "      DO" in code)
+    assert (
+        "      END DO\n" +
+        "      !$omp end taskloop\n" +
+        "      !$omp end master\n" +
+        "      !$omp end parallel" in code)
+
+    assert taskloop_node.begin_string() == "omp taskloop{0}".format(clauses)
+
+
+def test_omp_taskloop_apply_gencode():
+    '''Check that the gen_code method in the OMPTaskloopDirective
+    class generates the expected code when passing options to
+    the OMPTaskloopTransformation apply. Use the gocean API.
+    '''
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH, "single_invoke.f90"),
```

Comment:
If I understand correctly, this test is checking that the options passed to the `apply` method are passed on to the directive that is created. As such I think this test belongs in with those testing the transformation rather than the directive.

Reply:
Moved to transformations_test.